### PR TITLE
Update namespace for operator CRDs

### PR DESCRIPTION
**Pull request by alexellis**
## Description

* Update the namespace for operator's CRDs
* Use secrets for the license and auth token instead of flags in the deployment
* Make the operator also use a secret for the license
* Sync the license into namespaces where tunnels are created
* Move more fields from spec into status

## How Has This Been Tested?

All tests were run against DigitalOcean.

1. With a custom Tunnel YAML file

```yaml
apiVersion: operator.inlets.dev/v1alpha1
kind: Tunnel
metadata:
  name: gateway-tunnel
  namespace: openfaas
spec:
  serviceRef:
    name: gateway
    namespace: openfaas
status: {}
```

2. With a custom tunnel YAML file with a pre-existing auth token

`kubectl create secret generic -n openfaas   gateway-tunnel-token --from-literal token=XYZ`

```yaml
apiVersion: operator.inlets.dev/v1alpha1
kind: Tunnel
metadata:
  name: gateway-tunnel
  namespace: openfaas
spec:
  authTokenRef:
    name: gateway-tunnel-token
    namespace: openfaas
  serviceRef:
    name: gateway
    namespace: openfaas
status: {}
```

3. With exposing a Deployment

```bash
kubectl expose -n openfaas-fn deploy/figlet --type=LoadBalancer --name f1
```

```bash
kubectl get tunnel -n openfaas-fn -w
NAME        SERVICE   TUNNEL   HOSTSTATUS     HOSTIP   HOSTID      UPDATESERVICEIP
f1-tunnel   f1                 provisioning            342319775   true
openfaas-fn   f1-tunnel        f1                 active         143.110.174.140   342319775   true
f1-tunnel   f1                 active         143.110.174.140   342319775   true
f1-tunnel   f1                 active         143.110.174.140   342319775   true
openfaas-fn   f1-tunnel        f1                 active         143.110.174.140   342319775   true
```

```bash
kubectl get deploy -n openfaas-fn f1-tunnel-client
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
f1-tunnel-client   1/1     1            1           57s

kubectl get secret -n openfaas-fn
NAME             TYPE     DATA   AGE
f1-tunnel        Opaque   1      2m27s
inlets-license   Opaque   1      116m

> curl -d tunnel 143.110.174.140:8080
 _                          _ 
| |_ _   _ _ __  _ __   ___| |
| __| | | | '_ \| '_ \ / _ \ |
| |_| |_| | | | | | | |  __/ |
 \__|\__,_|_| |_|_| |_|\___|_|
                              
> 
```

In 3. I deleted the tunnel and saw it re-created, with a new tunnel server VM being provisioned.

I also tried deleting the tunnel in 1 and 2, and as expected, the VM was removed and nothing else happened after that.

I also tested that inlets can play nice with other LBs, with its --annotated-only flag.

With annotatedOnly set in the Helm chart, I didn't see a tunnel for the service, until I ran:
 
```
kubectl annotate -n openfaas-fn svc/f1 operator.inlets.dev/manage=1 --overwrite=1
service/f1 annotated
```

Later, setting this to 0, deleted the tunnel:

```
kubectl annotate -n openfaas-fn svc/f1 operator.inlets.dev/manage=0 --overwrite=1
service/f1 annotated
```

## How are existing users impacted? What migration steps/scripts do we need?

The namespace works out incorrectly to:

inlets.inlets.dev

This breaking change renames it to:

operator.inlets.dev

Any existing Tunnels would need to be migrated to the new namespace by exporting the contents and re-creating them under the new name.

The format of the CRD has changed to move more fields into the status that were previously located in the spec.

To test the changes, see the instructions in the Helm chart folder.

Build a custom image with:

```
PLATFORM=linux/amd64 SERVER=ttl.sh make push
```

Then update the values.yaml file and install the chart.

### CODE REVIEW

File: chart/inlets-operator/README.md [hunk 1]
```diff
@@ -17,19 +17,33 @@ If you would like to use Helm instead, read on.
...
 
     ```sh
-    kubectl create secret generic inlets-access-key \
-    --from-literal inlets-access-key="$(cat ~/Downloads/do-access-token)"
+    kubectl create secret generic -n inlets \
+    inlets-access-key \
+    --from-file inlets-access-key=$HOME/Downloads/do-access-token
     ```
 
 * Install the CRD:
```
Comment (welteki) [inline chart/inlets-operator/README.md:43]: We could also consider letting helm install the CRD instead of this manual step?

**Comment by welteki** [comment]
Tested these changes and looks like everything is working as expected.

**Review by welteki — APPROVED** [review]

File: chart/inlets-operator/README.md (hunk 1 above)
Comment (alexellis) [reply chart/inlets-operator/README.md:43]: What about managing changes in the CRD?

Comment (welteki) [reply chart/inlets-operator/README.md:43]: I think upgrading the CRD would still have to be done manually as Helm does not support that.

**Pull request merged**